Mathlib/Data/Set/Subset.lean

```diff
@@ -0,0 +1,346 @@
+
+@[simp]
+lemma restrict_subset_restrict_iff: A ↓∩ B ⊆ A ↓∩ C ↔ A ∩ B ⊆ A ∩ C := by
+  constructor
+  · rintro h x ⟨hxA,hxB⟩
+    constructor
+    · exact hxA
+    · specialize h ?_
+      · exact ⟨x,hxA⟩
+      · exact hxB
+      exact h
+  · rintro h ⟨x,hxA⟩ hx
+    specialize h ?_
+    · exact x
+    · exact ⟨hxA,hx⟩
+    exact h.2
+
+@[simp]
+lemma set_restrict_eq_iff :  A ↓∩ B = A ↓∩ C ↔ A ∩ B = A ∩ C  := by
+  simp only [subset_antisymm_iff, restrict_subset_restrict_iff, subset_inter_iff,
+    inter_subset_right, and_true]
+
+@[simp]
+lemma set_restrict_diff : A ↓∩ (B \ C) = (A ↓∩ B) \ (A ↓∩ C) := by
+  apply Eq.refl
+
+@[simp]
+lemma set_restrict_sUnion: A ↓∩ (⋃₀ S) = ⋃₀ {(A ↓∩ B) | B ∈ S} := by
+  ext x
+  simp only [mem_sUnion, mem_setOf_eq, exists_exists_and_eq_and, mem_set_restrict_iff]
+
+@[simp]
+lemma set_restrict_iUnion: A ↓∩ (⋃ (B : β ), i B) = ⋃ (B : β), A ↓∩ i B := by
+  ext x
+  simp only [mem_iUnion, mem_set_restrict_iff]
+
+@[simp]
+lemma set_restrict_iInter: A ↓∩ (⋂ (B : β ), i B) = ⋂ (B : β), A ↓∩ i B   := by
+  ext x
+  simp only [mem_iInter, mem_set_restrict_iff]
+
+@[simp]
+lemma set_restrict_sInter:  A ↓∩ (⋂₀ S)  = ⋂₀ {(A ↓∩ B) | B ∈ S} := by
+  ext x
+  simp only [mem_sInter, mem_setOf_eq, forall_exists_index, and_imp, forall_apply_eq_imp_iff₂,
+    mem_set_restrict_iff]
+
+lemma eq_of_restrict_eq_of_subset (hB : B ⊆ A) (hC : C ⊆ A) (h : A ↓∩ B = A ↓∩ C) : B = C := by
+  simp only [← inter_eq_right] at hB hC
+  simp only [set_restrict_eq_iff,hB,hC] at h
+  exact h
+
+lemma restrict_mono (h : B ⊆ C) : A ↓∩ B ⊆ A ↓∩ C := by
+  simp only [restrict_subset_restrict_iff, subset_inter_iff, inter_subset_left, true_and]
+  apply subset_trans (inter_subset_right A B) h
+
+lemma mem_coe_iff (x : α): x ∈  (↑D : Set α) ↔ ∃ y : ↑A, y ∈ D ∧ ↑y = x  := by rfl
+
+/--
+The following simp lemmas try to transform operations in the subtype into operations  in the ambient
+type, if possible.
+-/
+
+lemma coe_univ : ↑(univ : Set A)  = A := by
+  simp only [image_univ, Subtype.range_coe_subtype, setOf_mem_eq]
+
+lemma coe_empty : ↑(∅ : Set A)  = (∅ : Set α ) := image_empty _
+
+@[simp]
+lemma coe_union : (↑(D ∪ E) : Set α)  = ↑D ∪ ↑E := by
+  ext x
+  simp_all only [mem_union, mem_image, Subtype.exists, exists_and_right, exists_eq_right]
+  apply Iff.intro
+  · intro a
+    aesop_destruct_products
+    simp_all only [exists_true_left]
+  · intro a
+    cases a
+    · aesop_destruct_products
+      simp_all only [true_or, exists_const]
+    · aesop_destruct_products
+      simp_all only [or_true, exists_const]
+
+@[simp]
+lemma coe_inter : (↑(D ∩ E) : Set α) = ↑D  ∩ ↑E := by
+  ext
+  simp_all only [mem_inter_iff, mem_image, Subtype.exists, exists_and_right, exists_eq_right]
+  apply Iff.intro
+  · intro a
+    aesop_destruct_products
+    simp_all only [and_self, exists_const]
+  · intro a
+    aesop_destruct_products
+    simp_all only [exists_const, and_self]
+
+@[simp]
+lemma coe_compl : ↑(Dᶜ) = A \ ↑D := by
+  unhygienic ext
```

Comment:
At the end, all these `unhygienic` should be removed in favour of manually declared variable names.
```suggestion
  ext x
```

Reply:
Fixed